rotations.py

```diff
@@ -0,0 +1,415 @@
+
+    """
+    assert len(rot_vec) == 3
+    rot_vec = regularize_rotation_vector(rot_vec)
+
+    angle = np.linalg.norm(rot_vec)
+    skew_rot_vec = skew_matrix_from_vector(rot_vec)
+
+    if angle < epsilon:
+        sin_angle = 1 - (angle ** 2) / 6
+        cos_angle = 1 / 2 - angle ** 2
+    else:
+        sin_angle = np.sin(angle) / angle
+        cos_angle = (1 - np.cos(angle)) / (angle ** 2)
+
+    rot_mat = (np.identity(3) + sin_angle * skew_rot_vec
+               + cos_angle * np.dot(skew_rot_vec, skew_rot_vec))
+    return rot_mat
+
+
+# -- Group composition written in rotation vector
+
+
+def compose(rot_vec_1, rot_vec_2):
+    """
+    Compose 2 rotation vectors according to the matrix product
+    on the corresponding matrices.
+    """
+    rot_vec_1 = regularize_rotation_vector(rot_vec_1)
+    rot_vec_2 = regularize_rotation_vector(rot_vec_2)
+
+    rot_mat_1 = rotation_matrix_from_rotation_vector(rot_vec_1)
+    rot_mat_2 = rotation_matrix_from_rotation_vector(rot_vec_2)
+
+    rot_mat_prod = np.matmul(rot_mat_1, rot_mat_2)
+    rot_vec_prod = rotation_vector_from_rotation_matrix(rot_mat_prod)
+
+    return rot_vec_prod
+
+# -- Jacobians of left and right translations in SO(3)
+
+
+def jacobian_translation(rot_vec,
+                         left_or_right='left', epsilon=1e-5):
+    """
+    Compute the jacobian matrix of the differential
+    of the left translation by the rotation r.
+
+    :param rot_vec: 3D rotation vector
+    :returns jacobian: 3x3 matrix
+    """
+    assert len(rot_vec) == 3
+    rot_vec = regularize_rotation_vector(rot_vec)
+
+    angle = np.linalg.norm(rot_vec)
+    if angle < epsilon:
+        coef_1 = 1 - angle ** 2 / 12
+        coef_2 = 1 / 12 + angle ** 2 / 720
+    elif abs(angle - np.pi) < epsilon:
+        coef_1 = angle * (np.pi - angle) / 4
+        coef_2 = (1 - coef_1) / angle ** 2
+    else:
+        coef_1 = (angle / 2) / np.tan(angle / 2)
+        coef_2 = (1 - coef_1) / angle ** 2
+
+    if left_or_right == 'left':
+        jacobian = (coef_1 * np.identity(3)
+                    + coef_2 * np.outer(rot_vec, rot_vec)
+                    + skew_matrix_from_vector(rot_vec) / 2)
+
+    elif left_or_right == 'right':
+        jacobian = (coef_1 * np.identity(3)
+                    + coef_2 * np.outer(rot_vec, rot_vec)
+                    - skew_matrix_from_vector(rot_vec) / 2)
+    else:
+        raise ValueError('Param \'left_or_right\' '
+                         'should be \'left\' or \'right\'.')
+
+    return jacobian
+
+
+# -- Riemannian
+
+
+def riemannian_exp(tangent_vec,
+                   inner_product=ALGEBRA_CANONICAL_INNER_PRODUCT,
+                   left_or_right='left',
+                   ref_point=GROUP_IDENTITY):
+    """
+    Compute the Riemannian exponential at point ref_point
+    of tangent vector tangent_vec wrt the metric obtained by
+    left/right translation
+    of the inner product inner_product at the Lie algebra.
+
+    Formula:
+    R*Exp(DL(R^-1)*a)
```

Comment:
you can use latex in these comments

Reply:
Isn't it clearer like this?

Reply:
it depends if you generate the doc with readthedocs or something, you will get latex for free generated...

Reply:
Good to know! thanks.